services/s3/src/test/java/software/amazon/awssdk/services/s3/ExpiresHeaderDataTypeErrorTest.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.services.s3;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.any;
+import static com.github.tomakehurst.wiremock.client.WireMock.anyUrl;
+import static com.github.tomakehurst.wiremock.client.WireMock.stubFor;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.github.tomakehurst.wiremock.junit5.WireMockRuntimeInfo;
+import com.github.tomakehurst.wiremock.junit5.WireMockTest;
+import java.io.IOException;
+import java.net.URI;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+import software.amazon.awssdk.auth.credentials.AnonymousCredentialsProvider;
+
+@WireMockTest
+public class ExpiresHeaderDataTypeErrorTest {
+
+    private final String TEST_DATE = "2034-02-01T00:00:00Z";
+
+    public S3Client getS3Client(WireMockRuntimeInfo wm) {
+        return S3Client.builder().endpointOverride(URI.create(wm.getHttpBaseUrl()))
+                                  .credentialsProvider(AnonymousCredentialsProvider.create())
+                                  .build();
+    }
+
+    @Test
+    public void headObjectRequestWithInvalidDate_doesNotThrowException(WireMockRuntimeInfo wm) throws IOException {
+
+        stubFor(any(anyUrl())
+                    .willReturn(aResponse()
+                                    .withHeader("Expires", TEST_DATE)
+                                    .withBody("Hello world!")));
+
+        S3Client s3Client = getS3Client(wm);
```

Comment:
for Wiremock tests, I think you can use `@BeforeEach` with `WireMockRuntimeInfo`:

```java
private S3Client s3Client;

@BeforeEach
void init(WireMockRuntimeInfo wm) {
   this.s3Client = ...;
}
```

This would avoid having to call `getS3Client(wm)` manually in every test case.